src/components/text-input/Label.root.tsx

```diff
@@ -0,0 +1,74 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { applyTextStyles } from '../text';
+import { setAlpha, Utils } from '../../shared';
+import { applyMargin, applyPadding } from '../../utilities/layout';
+import { TTheme } from '../../utilities/theme';
+import Spacing from '../../utilities/spacing';
+
+import { TEXT_INPUT_VALUES_MAPPING } from './TextInput.constants';
+import { PLabelRoot } from './TextInput.props';
+import { TTextInputSizeToken } from './TextInput.types';
+
+const LabelRoot = styled.label<PLabelRoot>(
+    ({
+        theme: { background, palette },
+        active,
+        size,
+        value,
+        animatedLabel,
+        hasError,
+        leadingIcon,
+        backgroundColor = background.default,
+    }: ThemedStyledProps<PLabelRoot, TTheme>): FlattenSimpleInterpolation => {
+        const hasValue = Utils.isString(value) && value.length > 0;
+
+        const colors: Record<string, string> = {
+            active: hasError ? palette.alert.main : palette.primary.main,
+            background: backgroundColor || background.shape,
```

Comment:
backgroundColor cannot be undefined or falsey, therefore it will never be `background.shape` ... also you did assign a different default value in props destructuring

Reply:
Just saw this comment. I actually didn't push this change :D I will fix now